prboom2/src/z_zone.c

```diff
@@ -163,24 +156,23 @@ void Z_FreeTag(int tag)
 {
   memblock_t *block, *end_block;
 
-#ifdef HEAPDUMP
-  Z_DumpMemory();
-#endif
-
   if (tag < 0 || tag >= PU_MAX)
     I_Error("Z_FreeTag: Tag %i does not exist", tag);
 
-  block = blockbytag[tag];
-  if (!block)
-    return;
-  end_block = block->prev;
-  while (1)
+  for (; tag < PU_MAX; tag++)
```

Comment:
Why? There is no "order" to tags, so this would be an unexpected side effect.

Reply:
I understand that they happen to be in an order, and I understand that vanilla worked differently, but they do not have a guaranteed order in dsda-doom, it's incidental, and any one tag has no bearing on any other tag, so this operation has no value.

Reply:
I.e., let's say I introduced a tag for storing data related to graphics settings, and when the graphics are reinitialized we clear that tag. That would be nice and convenient, but how would it fit into the way you would clear tags?

Reply:
> This would also mean that clearing the level tag would unload all the unlocked lumps, so it would just degrade performance.

Yes, that's how it's done in, let's say, every other Doom source port. 😉 But I buy that argument about introducing another special-purpose tag that won't line up in the classic tag order.